src/core/sock/sockinfo_udp.cpp

```diff
@@ -1110,37 +1110,25 @@ int sockinfo_udp::setsockopt(int __level, int __optname, __const void *__optval,
             mc_if = mreqprm.imr_interface.s_addr;
 
             if (!IN_MULTICAST_N(mc_grp)) {
-                si_udp_logdbg(
-                    "setsockopt(%s) will be passed to OS for handling, IP %d.%d.%d.%d is not MC ",
-                    setsockopt_ip_opt_to_str(__optname), NIPQUAD(mc_grp));
+                si_udp_logdbg("setsockopt(%s) will be passed to OS for handling, IP %s is not MC ",
+                              setsockopt_ip_opt_to_str(__optname),
+                              ip_address(mc_grp).to_str(AF_INET).c_str());
                 break;
             }
 
-            // Find local interface IP address
+            // Find local interface for this MC ADD/DROP
             if (INADDR_ANY == mc_if) {
-                in_addr_t dst_ip = mc_grp;
-                in_addr_t src_ip = 0;
-
-                if ((!m_bound.is_anyaddr()) && (!m_bound.is_mc())) {
-                    src_ip = m_bound.get_ip_addr().get_in_addr();
-                } else if (!m_so_bindtodevice_ip.is_anyaddr()) {
-                    src_ip = m_so_bindtodevice_ip.get_in_addr();
-                }
-                // Find local if for this MC ADD/DROP
-                route_result res;
-                g_p_route_table_mgr->route_resolve(
-                    route_rule_table_key(ip_address(dst_ip), ip_address(src_ip), AF_INET, m_tos),
-                    res);
-                mc_if = res.src.get_in_addr();
-                si_udp_logdbg(
-                    "IPPROTO_IP, %s=%d.%d.%d.%d, mc_if:INADDR_ANY (resolved to: %d.%d.%d.%d)",
-                    setsockopt_ip_opt_to_str(__optname), NIPQUAD(mc_grp), NIPQUAD(mc_if));
-            } else {
-                si_udp_logdbg("IPPROTO_IP, %s=%d.%d.%d.%d, mc_if:%d.%d.%d.%d mc_src:%d.%d.%d.%d",
-                              setsockopt_ip_opt_to_str(__optname), NIPQUAD(mc_grp), NIPQUAD(mc_if),
-                              NIPQUAD(mreqprm.imr_sourceaddr.s_addr));
+                ip_address resolved_ip;
+                resolve_if_ip(0, ip_address(mc_grp), resolved_ip);
```

Comment:
see line 1565 in sockinfo_udp.cpp, we should probably use m_family there:
const auto &iparray = device_val->get_ip_array(AF_INET6);

instead AF_INET6

Reply:
Good catch

Reply:
Done